debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -127,13 +130,23 @@ public R apply(R r) {
             }
         }));
 
+        isDeleteEvent = payload == "" || payload == null || payload == "null";
```

Comment:
I think the "null" string shouldn't be interpreted here. Event the empty string is arguable, but it seems acceptable in case on NOT NULL column definition for the payload for whatever reasons.

Reply:
Yes I think we have to keep the empty string in the case where there is a NOT NULL constraint.